### PR TITLE
Changes beacon typepath

### PR DESCRIPTION
**Pull request by clusterfack**
Beacons use none of the advantages of being a radio, they don't send or receive signals in any portion of their code. The only thing they ever use from their inheritance is the variable name 'frequency' and NOTHING ELSE.

Therefore I've moved them under obj/item and gave them a list (because of course everywhere in the code used IN WORLD to find them).

### CODE REVIEW

**Comment by ririchiyo** [comment]
I'm unsure about this change.

**Comment by clusterfack** [comment]
Do you have any actual reason to be? All code that is affected is shown in the included files.

File: code/game/machinery/Beacon.dm [hunk 1]
```diff
@@ -9,12 +9,12 @@
...
 	use_power = 1
 	idle_power_usage = 0
-	var/obj/item/device/radio/beacon/Beacon
+	var/obj/item/beacon/Beacon
 
 	New()
 		..()
 		var/turf/T = loc
-		Beacon = new /obj/item/device/radio/beacon
+		Beacon = new /obj/item/beacon
```
Comment (ComicIronic) [inline code/game/machinery/Beacon.dm]: Hot damn, this entire file may be the worst code I've ever read.

File: code/game/objects/items/beacon.dm [hunk 2]
```diff
@@ -1,49 +1,58 @@
...
+
+	if ((usr.canmove && !( usr.restrained() )))
+		src.code = t
+	if (!( src.code ))
+		src.code = "beacon"
+	src.add_fingerprint(usr)
+	return
+
+
+/obj/item/beacon/bacon/proc/digest_delay()
```
Comment (ComicIronic) [inline code/game/objects/items/beacon.dm]: WHAT

Comment (clusterfack) [inline code/game/objects/items/beacon.dm]: I uh... I don't actually know. It was there before but it's not as if its used anywhere.

**Comment by Midaychi** [comment]
beacons be children of radios having frequencies is 100% legacy oldcode that no longer exists/has a use elsewhere.

**Pull request merged**